pkg/apis/kops/networking.go

```diff
@@ -47,3 +48,7 @@ type CNINetworkingSpec struct {
 // Kopeio declares that we want Kopeio networking
 type KopeioNetworkingSpec struct {
 }
+
+// Weave declares that we want Weave networking
+type WeaveNetworkingSpec struct {
```

Comment:
Looks good, but you need to copy and paste this field addition into the v1alpha1 package.  I typically copy & paste the whole file, then fix the package declaration